src/stores/FilterBarStore.js

```diff
@@ -0,0 +1,130 @@
+var SharedUtils = require('../utils/SharedUtils');
+
+export class FilterBarStore {
+  constructor(configuration) {
+    this.CHANGE_EVENT = 'change';
+    this.eventEmitter = new EventEmitter();
+
+    this.id = configuration.id;
+    this.persistent = configuration.persistent;
+    this.url = configuration.searchUrl;
+    this.searchUrl = configuration.searchUrl;
+    this.saveSearchUrl = configuration.saveSearchUrl;
+    this.savedSearchUrl = configuration.savedSearchUrl;
+    this.exportResultsUrl = configuration.exportResultsUrl;
+    this.filters = configuration.filters;
+
+    var filter, filterUid;
+
+    for (var i = 0; i < Object.keys(this.filters).length; i++) {
+      filterUid = Object.keys(this.filters)[i];
+      filter = this.filters[filterUid];
+
+      if (filter.url) {
+        SharedUtils.ajaxGet(filter.url, 'json', function(filterUid) { return function(response) { this.updateFilter(filterUid, 'options', response) } }(filterUid).bind(this));
+      }
+    }
+
+    this.receieveSavedSearches();
+  }
+
+  getId() {
+    return this.id;
+  }
+
+  getSearchUrl() {
+    return this.searchUrl;
+  }
+
+  getSaveSearchUrl() {
+    return this.saveSearchUrl;
+  }
+
+  getSavedSearches() {
+    return this.savedSearches;
+  }
+
+  getSavedSearch(searchId) {
+    return this.savedSearches[searchId];
+  }
+
+  getFilter(filterUid) {
+    return this.filters[filterUid]
+  }
+
+  getDisabled() {
+    var disabledFilters = {};
+    for (var filterUid in this.filters) {
+      if (this.filters[filterUid].enabled === false) {
+        disabledFilters[filterUid] = this.filters[filterUid];
+      }
+    }
+    return disabledFilters;
+  }
+
+  getEnabled() {
+    var enabledFilters = {};
```

Comment:
Except that this is working with objects, not arrays. I could loop over the Object.keys and use filter to return the filter?